src/lib/Zikula/Bundle/CoreInstallerBundle/Stage/Upgrade/NotUpgradedStage.php

```diff
@@ -55,9 +56,9 @@ public function isNecessary()
         // make sure selected language is installed
         $DBLocale = $this->fetchDBLocale();
         if (!in_array($DBLocale, \ZLanguage::getInstalledLanguages())) {
-            \System::setVar('language_i18n', 'en');
-            \System::setVar('language', 'eng');
-            \System::setVar('locale', 'en');
+            $this->container->get('zikula_extensions_module.api.variable')->set(VariableApi::CONFIG, 'language_i18n', 'en');
+            $this->container->get('zikula_extensions_module.api.variable')->set(VariableApi::CONFIG, 'language', 'eng');
+            $this->container->get('zikula_extensions_module.api.variable')->set(VariableApi::CONFIG, 'locale', 'en');
```

Comment:
Better assign the variable api to a local var first to avoid repeated method calls?